presto-main/src/main/java/com/facebook/presto/sql/planner/NodePartitioningManager.java

```diff
@@ -122,22 +126,21 @@ public NodePartitionMap getNodePartitioningMap(Session session, PartitioningHand
         ConnectorNodePartitioningProvider partitioningProvider = partitioningProviders.get(partitioningHandle.getConnectorId().get());
         checkArgument(partitioningProvider != null, "No partitioning provider for connector %s", partitioningHandle.getConnectorId().get());
 
-        Map<Integer, Node> bucketToNode = partitioningProvider.getBucketToNode(
-                partitioningHandle.getTransactionHandle().orElse(null),
-                session.toConnectorSession(),
-                partitioningHandle.getConnectorHandle());
-        checkArgument(bucketToNode != null, "No partition map %s", partitioningHandle);
-        checkArgument(!bucketToNode.isEmpty(), "Partition map %s is empty", partitioningHandle);
-
-        int bucketCount = bucketToNode.keySet().stream()
-                .mapToInt(Integer::intValue)
-                .max()
-                .getAsInt() + 1;
-
+        ConnectorBucketNodeMap connectorBucketNodeMap = getConnectorBucketNodeMap(session, partitioningHandle);
         // safety check for crazy partitioning
-        checkArgument(bucketCount < 1_000_000, "Too many buckets in partitioning: %s", bucketCount);
+        checkArgument(connectorBucketNodeMap.getBucketCount() < 1_000_000, "Too many buckets in partitioning: %s", connectorBucketNodeMap.getBucketCount());
+
+        Map<Integer, Node> bucketToNode = null;
+        if (connectorBucketNodeMap.hasFixedMapping()) {
```

Comment:
What does `fixed mapping` mean?

Reply:
@nezihyigitbasi  Connector has preference about the node placement of the bucket. This happens when Presto runs with a the shared-nothing style connector (e.g. Raptor)